src/execution/Canceller.ts

```diff
@@ -0,0 +1,48 @@
+import { promiseWithResolvers } from '../jsutils/promiseWithResolvers.js';
+
+/**
+ * A Canceller object that can be used to cancel multiple promises
+ * using a single AbortSignal.
+ *
+ * @internal
+ */
+export class Canceller {
```

Comment:
NIT: kind of dislike the naming, it's basically a promiseTracker class

Reply:
Renamed to PromiseCanceller….. any better?

Reply:
Actually, considering that the next step is to hook up early async iterator return to the abort signal, I think PromiseTracker/Canceller won't be accurate ... naming things so hard ...

What this class really does is allow us the ability to set lots of contingent things to be triggered by the AbortSignal without ever hitting a max event listener warning.

I'm going to merge it as we have it for now, but be very open to additional changes. :)